kernel/nic_buffers/src/lib.rs

```diff
@@ -29,17 +29,26 @@ impl TransmitBuffer {
             EntryFlags::WRITABLE | EntryFlags::NO_CACHE | EntryFlags::NO_EXECUTE,
         )?;
         Ok(TransmitBuffer {
-            mp: mp,
+            mp,
             phys_addr: starting_phys_addr,
             length: size_in_bytes,
         })
     }
 
-    // / Send this `TransmitBuffer` out through the given `NetworkInterfaceCard`. 
-    // / This function consumes this `TransmitBuffer`.
-    // pub fn send<N: NetworkInterfaceCard>(self, nic: &mut N) -> Result<(), &'static str> {
-    //     nic.send_packet(self)
-    // }
+    pub fn as_slice(&self) -> &[u8] {
+        // We checked that the mapped pages are >= to self.length during initialisation.
+        // There can be no overflows since length is a u16, nor can there be alignment
+        // issues because we are operating on u8s.
+        self.mp.as_slice(0, self.length.into()).unwrap() 
+    }
+    
+    pub fn as_slice_mut(&mut self) -> &mut [u8] {
+        // We checked that the mapped pages are >= to self.length during initialisation
+        // and that they are writable. There can be no overflows since length is
+        // a u16, nor can there be alignment issues because we are operating on
+        // u8s.
+        self.mp.as_slice_mut(0, self.length.into()).unwrap()
+    }
```

Comment:
same comments apply here as with `ReceiveBuffer` -- we must ensure the inner `MappedPages` isn't accessible. Again, I don't think it is but we ought to double check and then explicitly state it in the comments.

Reply:
should be addressed in https://github.com/theseus-os/Theseus/pull/677/commits/e5dde86deb35d9639afa89bb4d7ba95d207630cd